torch/csrc/distributed/rpc/python_functions.cpp

```diff
@@ -0,0 +1,65 @@
+#include <torch/csrc/distributed/rpc/python_functions.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+py::object to_py_obj(const Message& message) {
+  switch (message.type()) {
+    case MessageType::SCRIPT_RET: {
+      ScriptRet ret = ScriptRet::fromMessage(message);
+      Stack stack;
+      stack.push_back(ret.value());
+      return torch::jit::createPyObjectForStack(std::move(stack));
+    }
+    case MessageType::PYTHON_RET: {
+      return PythonRpcHandler::loadPythonUDFResult(message);
```

Comment:
Can we raise if the object is of `ExceptionBase` type?
Just like what Python `concurrent.futures.Future.results()` does.

Ref
https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.result



Otherwise, I have to unwrap the exception myself.
```Python
    fut = dist.rpc(func, inputs, kwargs, async_call=True)
    val = fut.wait()
    if isinstance(val, BaseException):
        raise val
    return val
```

Reply:
will try to add one

Reply:
@xush6528 I checked codes again, I think normal exception will be propagated and raised to user by calling fut.wait(), in this case "val = fut.wait()" will fail.  

in which case "val = fut.wait()" succeed and val is exception object? this looks like it is not system exception, but user anticipated exception?

Reply:
@zhaojuanmao 


```Python
# rpc.py

class RPCException(Exception):
    pass
```

```Python
# user_code.py

def raise_func():
    raise Exception("This is user exception")

fut = dist.rpc(to="worker1", raise_func, async_call=True)

try:
    return fut.wait()
except dist.RPCException as e:
    logger.error("RPC system failed.")
except BaseException:
    logger.error("User func failed to finish.")
```

Reply:
if user call raise exception on server side, server will crash, so we need remote exception handling and return remote exception as a new type of message and raise properly on client side. this is needed for both builitin operator and python UDF, I think it is in follow up diff list